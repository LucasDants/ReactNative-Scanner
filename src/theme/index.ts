export default {
  colors: {
    red: {
      900: '#B83341',
      800: '#E03F50',
    },
    blue: '#0000FF',
    white: '#FFFFFF',
    gray: {
      100: '#DCDCDC',
      600: '#404040',
      900: '#030303',
    },
  },
};