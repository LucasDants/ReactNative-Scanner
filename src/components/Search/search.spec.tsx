import React from 'react'

import { fireEvent, render } from '@testing-library/react-native'

import { ThemeProvider } from 'styled-components/native'
import theme from '../../theme'

import { Search } from '.'

const Providers = ({ children }) => (
  <ThemeProvider theme={theme}>
    { children }
  </ThemeProvider>
)

describe('Search Component', () => {

  it("should match the snapshot test", () => {
    const { toJSON } = render(
      <Search onClear={() => {}} onSearch={() => {}} />, {wrapper: Providers}
    )

    const component = toJSON()
    expect(component).toMatchSnapshot()

  })

  it("should have the correct value in the input", () => {
    const { getByTestId } = render(
       <Search onClear={() => {}} onSearch={() => {}} value="test-value" />, {wrapper: Providers}
    )

    const input = getByTestId("input")

    expect(input.props.value).toBe('test-value')

  })

  it("should call the onClear function when the button Clear is press", () => {
    const onClear = jest.fn();

    const { getByTestId } = render(
       <Search onClear={onClear} onSearch={() => {}} />, {wrapper: Providers}
    )

    const buttonClear = getByTestId("button-clear")

    fireEvent.press(buttonClear)

    expect(onClear).toBeCalledTimes(1)

  })

  it("should call the onSearch function when the button Search is press", () => {
    const onSearch = jest.fn();

    const { getByTestId } = render(
       <Search onClear={() => {}} onSearch={onSearch} />, {wrapper: Providers}
    )

    const buttonSearch = getByTestId("button-search")

    fireEvent.press(buttonSearch)

    expect(onSearch).toBeCalledTimes(1)

  })

})