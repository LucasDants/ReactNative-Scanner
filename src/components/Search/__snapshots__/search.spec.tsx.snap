// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component should match the snapshot test 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 10,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#404040",
          "borderRadius": 6,
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      placeholder="Search QRCodes..."
      placeholderTextColor="#DCDCDC"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "height": 52,
            "paddingLeft": 12,
          },
        ]
      }
      testID="input"
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginRight": 7,
          "opacity": 1,
          "paddingBottom": 12,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 12,
        }
      }
      testID="button-clear"
    >
      <Text />
    </View>
  </View>
  <RNGestureHandlerButton
    activeOpacity={0.105}
    collapsable={false}
    onActiveStateChange={[Function]}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#B83341",
        "borderRadius": 6,
        "height": 52,
        "justifyContent": "center",
        "marginLeft": 7,
        "width": 52,
      }
    }
    testID="button-search"
    underlayColor="black"
  >
    <View
      style={
        Object {
          "backgroundColor": "black",
          "borderBottomLeftRadius": undefined,
          "borderBottomRightRadius": undefined,
          "borderRadius": 6,
          "borderTopLeftRadius": undefined,
          "borderTopRightRadius": undefined,
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <Text />
  </RNGestureHandlerButton>
</View>
`;
